### PR TITLE
Build and release amazon-eks-pod-identity-webhook image from latest c…

### PR DESCRIPTION
**Pull request by vivek-koppuru**
*Issue #, if available:*

*Description of changes:*

Adding image building and release logic for amazon-eks-pod-identity-webhook binary as we want to be able to pull the latest changes from here: https://github.com/aws/amazon-eks-pod-identity-webhook. Changes to deploy the binary will follow. Tests will be able to run as soon as the below PR gets merged.

This PR is dependent on the following PR: https://github.com/aws/eks-distro-prow-jobs/pull/36

/hold

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by micahhausler — APPROVED** [review]

File: projects/aws/amazon-eks-pod-identity-webhook/Makefile [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+ifeq ("$(CLONE_URL)","")
+	$(error No clone url was provided.)
+endif
+
+ifeq ("$(REPO)","")
+	$(error No repository name was provided.)
+endif
+
+ifeq ("$(GIT_COMMIT)","")
```
Comment (TerryHowe) [inline projects/aws/amazon-eks-pod-identity-webhook/Makefile:13]: would this default off the GIT_COMMIT file?

Comment (vivek-koppuru) [reply projects/aws/amazon-eks-pod-identity-webhook/Makefile:13]: Yup

**Pull request merged**